sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp

```diff
@@ -0,0 +1,895 @@
+  Eigen::Vector3f p1((*input_)[samples[1]].x, (*input_)[samples[1]].y, (*input_)[samples[1]].z);
+  Eigen::Vector3f p2((*input_)[samples[2]].x, (*input_)[samples[2]].y, (*input_)[samples[2]].z);
+  Eigen::Vector3f p3((*input_)[samples[3]].x, (*input_)[samples[3]].y, (*input_)[samples[3]].z);
+  Eigen::Vector3f p4((*input_)[samples[4]].x, (*input_)[samples[4]].y, (*input_)[samples[4]].z);
+  Eigen::Vector3f p5((*input_)[samples[5]].x, (*input_)[samples[5]].y, (*input_)[samples[5]].z);
+  Eigen::Vector3f p6((*input_)[samples[6]].x, (*input_)[samples[6]].y, (*input_)[samples[6]].z);
+  Eigen::Vector3f p7((*input_)[samples[7]].x, (*input_)[samples[7]].y, (*input_)[samples[7]].z);
+  Eigen::Vector3f p8((*input_)[samples[8]].x, (*input_)[samples[8]].y, (*input_)[samples[8]].z);
+  Eigen::Vector3f p9((*input_)[samples[9]].x, (*input_)[samples[9]].y, (*input_)[samples[9]].z);
+
+  Eigen::Vector3f common_helper_vec = (p1 - p0).cross(p1 - p2);
+  Eigen::Vector3f ellipse_normal = common_helper_vec.normalized();
+
+  // The same check is implemented in isSampleGood, so be sure to look there too
+  // if you find the need to change something here.
+  if (common_helper_vec.squaredNorm() < Eigen::NumTraits<float>::dummy_precision ())
+  {
+    PCL_ERROR ("[pcl::SampleConsensusModelEllipse3D::computeModelCoefficients] Sample points too similar or collinear!\n");
+    return (false);
+  }
+
+  // Definition of the local reference frame of the ellipse
+  Eigen::Vector3f x_axis = (p1 - p0).normalized();
+  Eigen::Vector3f z_axis = ellipse_normal.normalized();
+  Eigen::Vector3f y_axis = z_axis.cross(x_axis).normalized();
+  
+  // Create the transposed rotation matrix
+  Eigen::Matrix3f Rot;
+  Rot << x_axis(0), y_axis(0), z_axis(0),
+      x_axis(1), y_axis(1), z_axis(1),
+      x_axis(2), y_axis(2), z_axis(2);
+
+  // Convert the points to local coordinates
+  Eigen::Vector3f p0_, p1_, p2_, p3_, p4_, p5_, p6_, p7_, p8_, p9_;
+  p0_ = Rot.transpose() * (p0 - p0);
+  p1_ = Rot.transpose() * (p1 - p0);
+  p2_ = Rot.transpose() * (p2 - p0);
+  p3_ = Rot.transpose() * (p3 - p0);
+  p4_ = Rot.transpose() * (p4 - p0);
+  p5_ = Rot.transpose() * (p5 - p0);
+  p6_ = Rot.transpose() * (p6 - p0);
+  p7_ = Rot.transpose() * (p7 - p0);
+  p8_ = Rot.transpose() * (p8 - p0);
+  p9_ = Rot.transpose() * (p9 - p0);
+  
+  /**
+   * Eigen::VectorXf fit_ellipse(std::array<Eigen::Vector3f, 10>);
+   *
+   * Implementation of the manuscript "Direct Least Square Fitting of Ellipses", A. Fitzgibbon, M. Pilu and R. Fisher,
+   * IEEE Transactions on Pattern Analysis and Machine Intelligence, 21(5): 476–480, May 1999.
+   * - calculates the Conic model parameters of an ellipse
+   */
+
+  // xOy projections only
+  Eigen::VectorXf X(6);
+  X << p0_(0), p1_(0), p2_(0), p3_(0), p4_(0), p5_(0), p6_(0), p7_(0), p8_(0), p9_(0);
+  Eigen::VectorXf Y(6);
+  Y << p0_(1), p1_(1), p2_(1), p3_(1), p4_(1), p5_(1), p6_(1), p7_(1), p8_(1), p9_(1);
```

Comment:
The vectors are 6 elements big, but you put 10 elements into them

Reply:
I increased the number of samples yesterday, what a gross mistake... Thanks for spotting the problem!